File: Daily Coding/python/reverse_vowels.py
# Reverse only the vowels present in the string
def check(letter):
    vowels = ['a','e','i','o','u']
    letter.lower()
    if letter in vowels:
        return True
    return False

word = input("Enter the word: ")
i = 0
j = len(word) - 1
lst = list(word)
while i<j:
    if not check(lst[i]):
        i += 1
    elif not check(lst[j]):
        j -= 1
    else:
        temp = lst[i]
        lst[i] = lst[j]
        lst[j] = temp
        i += 1
        j -= 1
word = "".join(lst)
print(word)

File: Daily Coding/c/reverse_vowels.c
/*
Reverse only the vowels present in the string
*/
#include <stdio.h>
int check(char l)
{
    if (l=='a' || l=='e' || l=='i' || l=='o' || l=='u')
        return 1;
    return 0;
}
int main()
{
    char word[50], temp;
    int i, j, len;
    printf("Enter the word: ");
    scanf("%s",word);
    for(len=0; word[len]; len++);
    i = 0;
    j = len-1;
    while(i < j)
    {
        if (!check(word[i]))
            i++;
        else if(!check(word[j]))
            j--;
        else
        {
            temp = word[i];
            word[i] = word[j];
            word[j] = temp;
            i++;
            j--;
        }
    }
    printf("%s",word);
    return 0;
}
